tests/ui/components/filters/sortoptionscomponent.js

```diff
@@ -182,4 +183,60 @@ describe('sort options component', () => {
     const wrapper = mount(component);
     expect(wrapper.text()).toEqual('');
   });
+
+  it('uses the persisted sortBys on load', () => {
+    const opts = {
+      ...defaultConfig,
+      options: threeOptions
+    };
+    const setSortBys = jest.fn();
+    COMPONENT_MANAGER.core.setSortBys = setSortBys;
+    COMPONENT_MANAGER.core.storage.setWithPersist(StorageKeys.SORT_BYS, [ threeOptions[1] ]);
+    const component = COMPONENT_MANAGER.create('SortOptions', opts);
+    const wrapper = mount(component);
+    expect(setSortBys).toHaveBeenCalledTimes(0);
+    expect(wrapper.find('.yxt-SortOptions-optionSelector').at(2).getDOMNode().checked).toBeTruthy();
+  });
+
+  it('ignores persisted sortBys that do not match any options from config', () => {
+    const opts = {
+      ...defaultConfig,
+      options: threeOptions
+    };
+    COMPONENT_MANAGER.core.storage.setWithPersist(StorageKeys.SORT_BYS, [{
+      type: 'fake type!'
+    }]);
+    const component = COMPONENT_MANAGER.create('SortOptions', opts);
+    const wrapper = mount(component);
+    expect(wrapper.find('.yxt-SortOptions-optionSelector').at(0).getDOMNode().checked).toBeTruthy();
+  });
+
+  it('sets persisted sort option on back navigation', () => {
+    const opts = {
+      ...defaultConfig,
+      options: threeOptions
+    };
+    const component = COMPONENT_MANAGER.create('SortOptions', opts);
+    const wrapper = mount(component);
+    expect(wrapper.find('.yxt-SortOptions-optionSelector').at(0).getDOMNode().checked).toBeTruthy();
+    COMPONENT_MANAGER.core.storage.setWithPersist(StorageKeys.SORT_BYS, [threeOptions[2]]);
```

Comment:
Should we also be pushing the state to history?

Reply:
no, SortOptions doesn't care about the actual history state, just the HISTORY_POP_STATE event. (Also afaik jsdom still doesn't support navigation)